core/server/src/main/java/alluxio/worker/block/evictor/LIRSEvictor.java

```diff
@@ -0,0 +1,573 @@
+ * will be stored in LIR cache and the others in HIR cache. No HIR blocks will be moved to LIR cache
+ * because all blocks have equal IRR. Therefore, all blocks in LIR cache will never be removed which
+ * promise the hit rate is at least (number of LIR blocks)/(number of all blocks). while if these
+ * blocks cannot be held in the memory in this case, LRU will achieve zero hit rate. It's also
+ * verified with Spark kmeans that the hit rate of LIRS policy improves a lot than other policies.
+ * So if your workloads contain a lot of iterations like kmeans, LIRS policy is recommended for you.
+ * Besides, LIRS can achieve as good performance as LRU and LRFU in most of workloads.
+ */
+public final class LIRSEvictor extends AbstractEvictor {
+  private static final Logger LOG = LoggerFactory.getLogger(Constants.LOGGER_TYPE);
+
+  private static final int LINKED_HASH_MAP_INIT_CAPACITY = 200;
+  private static final float LINKED_HASH_MAP_INIT_LOAD_FACTOR = 0.75f;
+  private static final boolean LINKED_HASH_MAP_ACCESS_ORDERED = false;
+  private static final boolean UNUSED_MAP_VALUE = true;
+
+  // The percent of HIR blocks on each tier
+  private final double mHIRPercent;
+  // The percent of LIR blocks on each tier
+  private final double mLIRPercent;
+  private final Configuration mConfiguration;
+  // Map from the location of a StorageDir to its condition of used space
+  private Map<BlockStoreLocation, SpaceContainer> mSpaceManager =
+      new ConcurrentHashMap<BlockStoreLocation, SpaceContainer>();
+  /**
+   * Map from the the block id to the block size. Maintain the information of block size is because
+   * the space of HIR block or LIR block need to be reclaimed when
+   * {@link #onRemoveBlockByClient(long, long)} or {@link #onRemoveBlockByWorker(long, long)} is
+   * called.
+   */
+  private Map<Long, Long> mBlockIdToSize = new ConcurrentHashMap<Long, Long>();
+  /**
+   * Record the information of all LIR blocks. Pair<BlockStoreLocation, Long> identifies a block
+   * because a block may contain more than one ghost blocks on all tiers.
+   */
+  private Map<Pair<BlockStoreLocation, Long>, BlockLIRSInfo> mLIRCache =
+      Collections.synchronizedMap(new LinkedHashMap<Pair<BlockStoreLocation, Long>, BlockLIRSInfo>(
+          LINKED_HASH_MAP_INIT_CAPACITY, LINKED_HASH_MAP_INIT_LOAD_FACTOR,
+          LINKED_HASH_MAP_ACCESS_ORDERED));
+  // Record the information of all HIR blocks.
+  private Map<Pair<BlockStoreLocation, Long>, BlockLIRSInfo> mHIRCache =
+      Collections.synchronizedMap(new LinkedHashMap<Pair<BlockStoreLocation, Long>, BlockLIRSInfo>(
+          LINKED_HASH_MAP_INIT_CAPACITY, LINKED_HASH_MAP_INIT_LOAD_FACTOR,
+          LINKED_HASH_MAP_ACCESS_ORDERED));
+  /**
+   * Record the information of all temp blocks. Temp blocks are blocks moved from other tiers and
+   * still not accessed by any clients.
+   */
+  private Map<Pair<BlockStoreLocation, Long>, Boolean> mTmpCache = Collections.synchronizedMap(

```

Comment:
@uronce-cc When a block not in memory is read, the client will first promote the block to memory tier and then read the block in memory tier. Thus, `onMove` and 'onAccess' will be called in one read request. We put the blocks moved from other tiers in tmp cache when `onMove` is called and then move the block to HIR cache when `onAccess` is called. In this way, we guarantee the block will not be put into LIR cache only in one read request.